pzsd_bot/__main__.py

```diff
@@ -1,48 +1,214 @@
+import asyncio
 import logging
 import os
-from pathlib import Path
+import re
+from datetime import datetime
 
 import discord
-import yaml
+from discord import Intents
 from dotenv import load_dotenv
+from sqlalchemy import insert, select
+from sqlalchemy.exc import MultipleResultsFound
+from sqlalchemy.ext.asyncio import create_async_engine
+from sqlalchemy.sql.functions import sum as sql_sum
 
-config_path = Path(__file__).parent / 'config.yml'
-if config_path.exists():
-    with config_path.open() as f:
-        config = yaml.safe_load(f)
-else:
-    config = {}
-
-LOG_LEVEL = config.get('LOG_LEVEL', 'INFO')
+from pzsd_bot.model import ledger, pzsd_user
 
 load_dotenv()
 
-BOT_TOKEN = os.environ['BOT_TOKEN']
-GUILD_ID = os.environ['GUILD_ID']
+LOG_LEVEL = os.getenv("LOG_LEVEL", "INFO")
+BOT_TOKEN = os.environ["BOT_TOKEN"]
+GUILD_ID = os.environ["GUILD_ID"]
+POINTS_LOG_CHANNEL = int(os.environ["POINTS_LOG_CHANNEL"])
+POINT_MAX_VALUE = 9223372036854775807
+
+connection_str = "postgresql+asyncpg://{}:{}@{}:{}/{}".format(
+    os.getenv("PGUSER", "postgres"),
+    os.getenv("PGPASSWORD", "password"),
+    os.getenv("PGHOST", "localhost"),
+    os.getenv("PGPORT", "5432"),
+    os.getenv("PGDATABASE", "pzsd"),
+)
+engine = create_async_engine(connection_str)
 
-logger = logging.getLogger('pzsd')
+logger = logging.getLogger("pzsd")
 logger.setLevel(LOG_LEVEL)
-handler = logging.FileHandler(filename='pzsd_bot.log', encoding='utf-8', mode='w')
-handler.setFormatter(logging.Formatter('%(asctime)s:%(levelname)s:%(name)s: %(message)s'))
+handler = logging.FileHandler(filename="pzsd_bot.log", encoding="utf-8", mode="a")
+handler.setFormatter(
+    logging.Formatter("%(asctime)s:%(levelname)s:%(name)s: %(message)s")
+)
 logger.addHandler(handler)
 
-bot = discord.Bot()
+bot = discord.Bot(intents=Intents.all())
+
+point_pattern = re.compile(
+    r"(?:^| )(?P<point_amount>[+-]?(?:\d+|\d{1,3}(?:,\d{3})*)) "
+    r"+points? (?:to|for) (?:(?P<recipient_name>\w+)|<@(?P<recipient_id>\d+)>)",
+    re.IGNORECASE,
+)
 
 
 @bot.event
 async def on_ready():
-    print(f"We have logged in as {bot.user}")
+    print(f"Ready.")
```

Comment:
why

Reply:
also unnecessary f string

Reply:
Why have a `print` statement or why is it a format string? If the former, it's so there's something I can see to confirm the bot is actually running without having to check the logs. If the latter, I just forgot to remove the `f` when I changed the message.